library/dash/src/main/java/com/google/android/exoplayer2/source/dash/manifest/SegmentBase.java

```diff
@@ -102,6 +102,7 @@ public RangedUri getIndex() {
     /* package */ final long startNumber;
     /* package */ final long duration;
     /* package */ final List<SegmentTimelineElement> segmentTimeline;
+    /* package */ final int endNumber;
```

Comment:
* Is there a reason this is defined here rather than on `SegmentTemplate`? Given it's only used there, it seems simpler just to define it on that class directly?
* I think this needs to be a long rather than an int (given `startNumber` is a long and it must be true that `endNumber >= startNumber`. Ditto all the plumbing of `endNumber`

Reply:
Have moved endNumber to SegmentTemplate.